specs/directory.md

```diff
@@ -0,0 +1,387 @@
+| --------- | ----------------------------------------------------------------------------------- |
+| `BIN`     | Directory for executable binaries                                                   |
+| `LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `CONFIG`  | Directory for configurations                                                        |
+| `CACHE`   | Directory for user-specific cache                                                   |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.
+
+On Unix systems, `notation` follows [Filesystem Hierarchy Standard][FHS] for system level directories and [XDG Base Directory Specification][XDG] for user level directories. On Windows, [Known Folders][KF] and [App Settings][AS] are followed equivalently. On Darwin, [macOS File System][macOS_FS] with [System Integrity Protection][SIP] is followed equivalently. If a file with the same name exists at the system level and the user level, the file at the user level takes over the priority.
+
+### System Level
+
+Default directory paths for various operating systems at system level are specified as below.
+
+| Directory | Unix           | Windows                       | Darwin                         |
+| --------- | -------------- | ----------------------------- | ------------------------------ |
+| `BIN`     | `/usr/bin`     | `%ProgramFiles%/notation/bin` | `/usr/local/bin`               |
+| `LIBEXEC` | `/usr/libexec` | `%ProgramFiles%`              | `/usr/local/lib`               |
+| `CONFIG`  | `/etc`         | `%ProgramData%`               | `/Library/Application Support` |
+
+`CACHE` is omitted since it is user specific.
+
+### User Level
+
+Default directory paths for various operating systems at user level are specified as below.
+
+| Directory | Unix               | Windows          | Darwin                          |
+| --------- | ------------------ | ---------------- | ------------------------------- |
+| `LIBEXEC` | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CONFIG`  | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CACHE`   | `$XDG_CACHE_HOME`  | `%LocalAppData%` | `~/Library/Caches`              |
+
+There is no default `BIN` path at user level since the `notation` binary can be put anywhere as long as it in the `PATH` environment variable. Common directories on Unix/Darwin are `~/bin` and `~/.local/bin` where manual `PATH` update by users may be required.
+
+## Structure
+
+The overall directory structure for `notation` is summarized as follows.
+
+```
+{BIN}
+└── notation
+{CACHE}
+└── notation
+    └── signature
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}
+                    └── {signature-digest}.sig

```

Comment:
Is there a reason we need a sub directory for `{signature-digest-algorithm}`, a given manifest digest will only contain one signature. 
Is `{manifest-digest}` the artifact manifest digest or signature manifest digest, maybe update the dir structure to make that clearer.

Reply:
A manifest may have multiple signatures signed by various parties.

The `{manifest-digest}` is the digest of the subject manifest. i.e. The manifest to be signed.

Reply:
The section `Signature Caches` clarifies that the manifest is the manifest to be signed.